.changeset/lazy-zoos-move.md

```diff
@@ -0,0 +1,67 @@
+---
+'@backstage/plugin-catalog-backend-module-gitlab': patch
+---
+
+Add a new provider `GitlabDiscoveryEntityProvider` as replacement for `GitlabDiscoveryProcessor`
+
+In order to migrate from the `GitlabDiscoveryProcessor` you need to apply
+the following changes:
+
+**Before:**
+
+```yaml
+# app-config.yaml
+
+catalog:
+  locations:
+    - type: gitlab-discovery
+      target: https://company.gitlab.com/prefix/*/catalog-info.yaml
+```
+
+```ts
+/* packages/backend/src/plugins/catalog.ts */
+
+import { GitlabDiscoveryProcessor } from '@backstage/plugin-catalog-backend-module-gitlab';
+
+const builder = await CatalogBuilder.create(env);
+/** ... other processors ... */
+builder.addProcessor(
+  GitLabDiscoveryProcessor.fromConfig(env.config, { logger: env.logger }),
+);
+```
+
+**After:**
+
+```yaml
+# app-config.yaml
+
+catalog:
+  providers:
+    gitlab:
+      yourProviderId: # identifies your dataset / provider independent of config changes
+        host: gitlab-host # Identifies one of the hosts set up in the integrations
+        branch: main # Optional. Uses `master` as default
```

Comment:
Without having read the code yet - isn't this a per-repo setting? We already have `GitLabProject.default_branch` in this module

Reply:
Ah alright, that's just unfortunate then that we don't always get the requisite data easily at hand.

Reply:
@ivangonzalezacuna and @freben 

Em branch, consigo passar uma lista?
Exemplo:
```
branch: 
  - main
  - master
  - backstage
```

E posso ignorar o parametro group?

Thanks!